src/main/java/com/zendesk/maxwell/row/RowMap.java

```diff
@@ -39,7 +41,9 @@
 	private Long threadId;
 
 	private final LinkedHashMap<String, Object> data;
+	private final HashMap<String, ColumnDef> dataColumnDef;
 	private final LinkedHashMap<String, Object> oldData;
+	private final HashMap<String, ColumnDef> oldDataColumnDef;
```

Comment:
Is there any case where the column def in `old` and `data` would be different? If not, we could get rid of this.

Reply:
Good a point. It seems there no cases where the column def in `old` and `data` would be different. I hope one thing it would be nice in terms of API consistency if we could define `getOldDataColumnDef` method in `RowMap` class. Do you want me to get rid of that method as well? Or just want me to reuse `dataColumnDef` variable in `getOldDataColumnDef`/`putOldData`?

Reply:
A `RowMap` can't represent data _across_ a schema change, so we can guarantee that old + new data must be the same column def. So I think it's fine to just update the same map from both `putData` and `putOldData`, and have a single `getColumnDef` which accesses this map.

If you want to be paranoid you could assert that whenever we put a column def in the map we're either adding a new entry to the map or replacing an identical column def, but that should already be the case.